Create `zk-explorer` app, implement project submission page

yurixander opened the pull request:
## Summary of changes

_Provide a detailed description of proposed changes._

- Created and integrated the `zk-explorer` app into the monorepo, which is based in Next.js and utilizes a shared technology stack including Tailwind, React, etc. No new dependencies were added (#1822).
- Added `start:zk-explorer` and `build:zk-explorer` scripts to the monorepo's `package.json` file.
- Added `zk-explorer` to the `README` file along with a short description.
- Created basic `README` for `zk-explorer` project.
- Created responsive design implementation of the project submission page (#1825).
- Created `GitHubOAuthButton` component to be used for GitHub OAuth authentication flow.
- Created barebones `Header` component implementation, which includes the use of breadcrumbs, search, and GitHub OAuth sign in button.
- Added `NEXT_PUBLIC_GITHUB_CLIENT_ID` environment variable entry to `.env.example`, which helps specify the GitHub application's client ID to be used during the OAuth authentication flow.
- Added client-size GitHub project URL validation to the project submission input using Regex.

### Proposed area of change

_Put an `x` in the boxes that apply._

- [ ] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [ ] `apps/tangle-dapp`
- [ ] `apps/testnet-leaderboard`
- [ ] `apps/faucet`
- [x] `apps/zk-explorer`
- [ ] `libs/webb-ui-components`

### Reference issue to close (if applicable)

- Closes #1825

### Screen Recording

_If possible provide a screen recording of proposed change._

https://github.com/webb-tools/webb-dapp/assets/101931215/ef219322-300a-4823-ace4-98833738de9c

---

### Code Checklist

_Please be sure to add .stories documentation if any additions are made to `libs/webb-ui-components`._

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

vutuanlinh2k2 commented on apps/zk-explorer/public/favicon.ico:
remove empty file

vutuanlinh2k2 commented on apps/zk-explorer/public/.gitkeep:
remove empty file

vutuanlinh2k2 commented on apps/zk-explorer/app/page.tsx:
```diff
@@ -0,0 +1,94 @@
+  const [isValidGithubUrl, setIsValidGithubUrl] = useState(false);
+  const [isGithubUrlDisplayingError, setIsGithubUrlDisplayingError] = useState(false);
+
+  const validateGithubUrl = (url: string): boolean => {
```
consider move this into utils folder or utils.ts file

vutuanlinh2k2 commented on apps/zk-explorer/components/GitHubOAuthButton.tsx:
consider turn this into a folder and move all the types into types.ts file

vutuanlinh2k2 commented on apps/zk-explorer/components/Header.tsx:
```diff
@@ -0,0 +1,62 @@
+import { GitHubOAuthButton } from './GitHubOAuthButton';
+
+export const Header: FC<unknown> = () => {
+  const githubOAuthClientId = process.env.NEXT_PUBLIC_GITHUB_CLIENT_ID;
```
in next.config.js, follow this [code](https://github.com/webb-tools/webb-dapp/blob/develop/apps/hubble-stats/next.config.js#L15) to make NEXT_PUBLIC_GITHUB_CLIENT_ID accessible to the client

vutuanlinh2k2 reviewed (changes requested):
Great work @yurixander ⭐ ⭐ ⭐ 

The CI is currently broken. Rember to rub `yarn lint` and `yarn format` before pushing to the remote branch

yurixander commented:
Thanks for the feedback, @vutuanlinh2k2!

* I've organized the types, utility functions, and also removed the empty files as directed.
* Addressed the issue regarding the environment variable.
* I've setup a local Git pre-commit hook to depend on `yarn lint && yarn format:check` so that I don't forget again 😄

vutuanlinh2k2 commented on apps/zk-explorer/app/page.tsx:
```diff
@@ -0,0 +1,81 @@
+        <Card className="max-w-[780px]">
+          <Typography variant="h5">1. GitHub Repository URL:</Typography>
+
+          <Input
```
Consider creating a new component for this Input and moving all the hooks into it. We should handle client actions in components rather making the whole page client-side

vutuanlinh2k2 reviewed (changes requested):
Just one more thing that I think we should change, great work @yurixander

monaiuu reviewed (approved):
Looks good.

@AtelyPham can we setup netlify deployment preview for `zk-explorer`?
